src/handlers/tokens.js

```diff
@@ -0,0 +1,66 @@
+const Cache = require('node-cache');
+
+const CHUNK_SIZE = 25;
+
+const cache = new Cache({ stdTTL: 60 * 60, checkperiod: 0, useClones: false });
+
+function parseTokenNamesFromURI ({ tokens }) {
+    try {
+        const tokensCollection = decodeURIComponent(tokens).split(',');
+        if (!tokensCollection[0].length) {
+            throw new Error('Argument is not an array or being null');
+        }
+        return tokensCollection;
+    } catch (err) {
+        console.error(err);
+        return [];
+    }
+}
+
+function arrayPartition (tokens, size) {
+    let chunkedList = [];
+    for (let i = 0; i < tokens.length; i += size) {
+        chunkedList.push(tokens.slice(i, i + size));
+    }
+    return chunkedList;
+}
+
+async function execBatch (batches, callback) {
```

Comment:
I feel something odd about "execBatch" naming, when the first parameter is "batches" (multiple).
Feels a bit counter-intuitive
Can we come up with a better naming here?